### PR TITLE
Javascript/javascript2/week1

### PR DESCRIPTION
**Pull request by JyothiNandyala**
Javascript2 week1 homework files.

### CODE REVIEW

File: javascript/javascript2/week1/shortestword.js [hunk 1]
```diff
@@ -0,0 +1,18 @@
+
+const danishWords = ["bil", "plante", "kaffe", "bog", "ø", "planetarium"];
+function checkTheShortestWord(){
+    let shortestWord;
+    let shortestWordLength=danishWords[0].length;
+
+    for(let i=0;i<danishWords.length;i++){
```
Comment (Rassibassi) [inline javascript/javascript2/week1/shortestword.js:7]: In the first loop iteration your code will always compare the first word with itself!
E.g., here in the first iteration both word and shortestWord are "bil"
Which means there is always one unnecessary iteration. Can you change the for loop such that this is solved?

File: javascript/javascript2/week1/countdanishLetters.js [hunk 2]
```diff
@@ -0,0 +1,29 @@
+//const danishString = "Jeg har en blå bil";
+const danishLetters="å,ø,æ";
```
Comment (Rassibassi) [inline javascript/javascript2/week1/countdanishLetters.js:2]: This variable is never used

File: javascript/javascript2/week1/countdanishLetters.js [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+    returnObject={total:charå+charø+charæ ,å:charå,ø:charø,æ:charæ };
+        
+} 
+return returnObject;
+}
+const danishString = "Jeg har en blå bil";
+console.log(checkDanishLetters(danishString)); // returns {total: 1, å: 1}
+
+const danishString2 = "Blå grød med røde bær";
+console.log(checkDanishLetters(danishString2)); // returns {total: 4, æ: 1, ø: 2, å: 1}
```
Comment (Rassibassi) [inline javascript/javascript2/week1/countdanishLetters.js:29]: here, please properly indent your code:
```
function checkDanishLetters(string){
    let charå=0;
    let charø=0;
    let charæ=0;
    let returnObject={};
    for(let i=0;i<string.length;i++){
        let char=string[i];
        if(char==='å'){
            
            charå++;
        }
        if(char==='ø'){
            charø++;
        }
        if(char==='æ'){
            charæ++;
        }
        returnObject={total:charå+charø+charæ ,å:charå,ø:charø,æ:charæ };
        
    } 
    return returnObject;
}
```

File: javascript/javascript2/week1/countdanishLetters.js [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+    
+    charå++;
+}
+if(char==='ø'){
+    charø++;
+    }
+    if(char==='æ'){
+        charæ++;
+    }
+    returnObject={total:charå+charø+charæ ,å:charå,ø:charø,æ:charæ };
```
Comment (Rassibassi) [inline javascript/javascript2/week1/countdanishLetters.js:20]: this line can go outside the for loop

File: javascript/javascript2/week1/countdanishLetters.js [hunk 5]
```diff
@@ -0,0 +1,29 @@
+//const danishString = "Jeg har en blå bil";
+const danishLetters="å,ø,æ";
+function checkDanishLetters(string){
+    let charå=0;
+    let charø=0;
+    let charæ=0;
+    let returnObject={};
```
Comment (Rassibassi) [inline javascript/javascript2/week1/countdanishLetters.js:7]: why not `const`?

File: javascript/javascript2/week1/countdanishLetters.js [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+    for(let i=0;i<string.length;i++){
+        let char=string[i];
```
Comment (Rassibassi) [inline javascript/javascript2/week1/countdanishLetters.js:9]: why not `const`?

File: javascript/javascript2/week1/countdanishLetters.js [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+    if(char==='æ'){
+        charæ++;
+    }
+    returnObject={total:charå+charø+charæ ,å:charå,ø:charø,æ:charæ };
+        
+} 
+return returnObject;
+}
+const danishString = "Jeg har en blå bil";
+console.log(checkDanishLetters(danishString)); // returns {total: 1, å: 1}
```
Comment (Rassibassi) [inline javascript/javascript2/week1/countdanishLetters.js:26]: this line outputs:
`{ total: 1, 'å': 1, 'ø': 0, 'æ': 0 }` not `{total: 1, å: 1}`
How can you change your code to not show `0` entries?

File: javascript/javascript2/week1/animalNameGenerator.html [hunk 8]
```diff
@@ -0,0 +1,45 @@
...
+    <meta charset="UTF-8">
+    <meta name="viewport" content="width=device-width, initial-scale=1.0">
+    <title>SpiritAnimalNameGenarator</title>
+</head>
+<body>
+    <form>
+        <label for="name">Name of the user:</label>
+        <input type="text" name="Name" id="txtName" placeholder="Name"><br>
+        
+<input type="button" value="Clickme" id="btnSubmit" onclick="getUserName()">
```
Comment (Rassibassi) [inline javascript/javascript2/week1/animalNameGenerator.html:13]: here you put an `onclick` event and refer to the function `getUserName`, can you also do that with the `addEventListener` functionality?

File: javascript/javascript2/week1/animalNameGenerator.html [hunk 9]
```diff
@@ -0,0 +1,45 @@
...
+    <div>
+    <p id="spiritname">
+
+    </p>
+</div>
+<script>
+    let spritAnimalsString=["The fullmoon wolf","The crying butterfly","wak frog","moving the snake","crying the crow",
+"dancing peacock","The singing parrot","The eating cow","Talking pigeon","The fighting lion"];
+let randomElement ;
+const randomNumber=Math.floor(Math.random(randomElement)*10)+0;
```
Comment (Rassibassi) [inline javascript/javascript2/week1/animalNameGenerator.html:24]: if you create the `randomNumber` inside the `getUserName` function, your code will regenerate a new spirit animal everytime you click

File: javascript/javascript2/week1/spiritAnimalNameOptions.html [hunk 10]
```diff
@@ -0,0 +1,77 @@
...
+
+<body>
+    <form>
+        <label for="spirit">select the when spirit animal should be created.</label>
+        <br><br>
+        <label for="fname">Name Of The user:</label>
+        <input type="text" id="name"><br><br>
+        <br>
+
+        <select id="selectedOptions" name="selectedOptions" onchange="selectOptions()">
```
Comment (Rassibassi) [inline javascript/javascript2/week1/spiritAnimalNameOptions.html:18]: Use `addEventListener` in the Javascript to create this `onchange` event.

File: javascript/javascript2/week1/spiritAnimalNameOptions.html [hunk 11]
```diff
@@ -0,0 +1,77 @@
...
+    <script type="text/javascript">
+        
+    let spiritAnimalsString=["The fullmoon wolf","The crying butterfly","wak frog","moving the snake","crying the crow",
+"dancing peacock","The singing parrot","The eating cow","Talking pigeon","The fighting lion"];
+
+const button = document.getElementById("button");
+
+const result = document.getElementById("result");
+
+const randomSpiritAnimalArray = Math.floor(Math.random() * 10);
```
Comment (Rassibassi) [inline javascript/javascript2/week1/spiritAnimalNameOptions.html:41]: same here, move this variable into the `getUserName` function :)

**Review by Rassibassi — COMMENTED** [review]
Good work, let me know if you have any questions regarding my comments

**Comment by JyothiNandyala** [comment]
hi rasmus i made the changes ,could you please  let me know can i able to
merge the pull request.

**Comment by JyothiNandyala** [comment]
made the changes with mentor comments.

**Pull request merged**